core-web/libs/sdk/angular/src/lib/layout/dotcms-layout/dotcms-layout.component.spec.ts

```diff
@@ -153,8 +159,36 @@ describe('DotcmsLayoutComponent', () => {
             beforeEach(() => spectator.detectChanges());
 
             it('should update the page asset when changes are made in the editor', () => {
-                expect(client.editor.on).toHaveBeenCalledWith('changes', expect.any(Function));
+                const editorOnSpy = jest.spyOn(client.editor, 'on');
+                expect(editorOnSpy).toHaveBeenCalledWith('changes', expect.any(Function));
             });
         });
     });
+
+    describe('template', () => {
+        beforeEach(() => spectator.detectChanges());
+
+        it('should render rows', () => {
+            expect(spectator.queryAll(RowComponent).length).toBe(3);
+        });
+
+        it('should pass the correct row to RowComponent', () => {
+            const rowComponents = spectator.queryAll(RowComponent);
+            const rows = PageResponseMock.layout.body.rows;
+            expect(rowComponents.length).toBe(rows.length);
+            expect(rowComponents[0].row).toEqual(rows[0]);
+            expect(rowComponents[1].row).toEqual(rows[1]);
+            expect(rowComponents[2].row).toEqual(rows[2]);
```

Comment:
this can be a for loop to reduce code complexity

Reply:
Sure!